### ISSUE TITLE
Intel microcode breaks building of kernel on x86_64

### ISSUE THREAD

**Issue by AxelVoitier** [linuxkit/linuxkit#2894]
**Description**

The building of kernel on x86_64 is broken by latest Intel microcode being pulled out from their website.
It is most likely related to #2884. But I report it here more "formally" because it actually breaks compilation of the kernel on x86 system at the moment.

**Steps to reproduce the issue:**

```make build_4.14.x``` in the kernel folder.

**Describe the results you received:**

```
Step 26/33 : ENV UCODE_URL=https://downloadmirror.intel.com/27431/eng/microcode-20180108.tgz
 ---> Running in 79a655785f20
Removing intermediate container 79a655785f20
 ---> 0c5c695e42c6
Step 27/33 : RUN set -e &&     if [ $(uname -m) == x86_64 ]; then         cd /ucode &&         curl -sSL -o microcode.tar.gz ${UCODE_URL} &&         md5sum -c intel-ucode-md5sums &&         tar xf microcode.tar.gz &&         iucode_tool --normal-earlyfw --write-earlyfw=/out/intel-ucode.cpio ./intel-ucode &&         cp intel-ucode-license.txt /out;     fi
 ---> Running in 4436e1f8fdfd
microcode.tar.gz: FAILED
md5sum: WARNING: 1 of 1 computed checksums did NOT match
The command '/bin/sh -c set -e &&     if [ $(uname -m) == x86_64 ]; then         cd /ucode &&         curl -sSL -o microcode.tar.gz ${UCODE_URL} &&         md5sum -c intel-ucode-md5sums &&         tar xf microcode.tar.gz &&         iucode_tool --normal-earlyfw --write-earlyfw=/out/intel-ucode.cpio ./intel-ucode &&         cp intel-ucode-license.txt /out;     fi' returned a non-zero code: 1
make: *** [build_4.14.x] Erreur 1
```

So it complains about bad md5 checksum. But if you investigate further and try to download the microcode directly on the host, you find out Intel actually removed the microcode archive:

```
$ wget https://downloadmirror.intel.com/27431/eng/microcode-20180108.tgz
--2018-01-30 12:34:56--  https://downloadmirror.intel.com/27431/eng/microcode-20180108.tgz
Résolution de downloadmirror.intel.com (downloadmirror.intel.com)… 104.93.244.15
Connexion à downloadmirror.intel.com (downloadmirror.intel.com)|104.93.244.15|:443… connecté.
requête HTTP transmise, en attente de la réponse… 302 Moved Temporarily
Emplacement : http://www.intel.com/content/www/us/en/404.html [suivant]
--2018-01-30 12:34:57--  http://www.intel.com/content/www/us/en/404.html
Résolution de www.intel.com (www.intel.com)… 104.93.244.15
Connexion à www.intel.com (www.intel.com)|104.93.244.15|:80… connecté.
requête HTTP transmise, en attente de la réponse… 301 Moved Permanently
Emplacement : https://www.intel.com/content/www/us/en/404.html [suivant]
--2018-01-30 12:34:57--  https://www.intel.com/content/www/us/en/404.html
Connexion à www.intel.com (www.intel.com)|104.93.244.15|:443… connecté.
requête HTTP transmise, en attente de la réponse… 404 Not Found
2018-01-30 12:34:57 erreur 404 : Not Found.
```

Searching on Intel website for the microcode shows that the latest version is listed as 20171117.

**Describe the results you expected:**

A successful kernel compilation :).

**Additional information you deem important (e.g. issue happens only occasionally):**

Probably a fallout of Meltdown/Spectre mess. To this avail, what would be best? Follow Intel advice and rollback to previous microcode? Or keep applying their allegedly bogus microcode (I say allegedly because I have it installed on several systems of various generations and none are being affected by random reboot so far).

**Comment by rn** [comment]
Thanks for the report. I noticed last week that the orginal page was pulled but the kernel compile update last week worked fine. Looks like they now have also pulled the download link.

If it is not too urgent I will fix this as part of the next kernel update later this week.

**Comment by AxelVoitier** [comment]
Thanks. I don't think there is any urgency to it. If someone needs to compile a custom kernel they can probably figure out that commenting this part in the Dockerfile is enough to pass. And if they do need a previous microcode update, then it seems rather easy to fix themselves for 20171117 (just change the url and update the checksum).

For my part I was merely trying out the build process in order to test a custom kernel for a different arch. So I won't be affected by this ;).

**Comment by rn** [comment]
resolved via #2896
